snippets/base/admin/adminmodels.py

```diff
@@ -619,31 +620,50 @@ def get_form(self, request, obj=None, **kwargs):
         form.current_user = request.user
         return form
 
-    def make_published(self, request, queryset):
-        clean_queryset = queryset.exclude(status=models.STATUS_CHOICES['Published'])
+    def _action_status_change(self, action, request, queryset):
+        if action == 'publish':
+            status = models.STATUS_CHOICES['Published']
+            no_action_message = 'Skipped {} already published snippets.'
+            success_message = 'Published {} snippets.'
+        else:
+            status = models.STATUS_CHOICES['Draft']
+            no_action_message = 'Skipped {} already unpublished snippets.'
+            success_message = 'Unpublished {} snippets.'
+
+        clean_queryset = queryset.exclude(status=status)
         no_snippets = clean_queryset.count()
         no_already_published_snippets = queryset.count() - no_snippets
 
-        snippets = []
-        with transaction.atomic():
-            for snippet in clean_queryset:
-                snippet.status = models.STATUS_CHOICES['Published']
-                snippet.save()
-                snippets.append(snippet)
+        now = datetime.utcnow()
+        # Create a list of matching snippets before altering them to log action
+        # and send Slack messages.
+        snippets = [snippet for snippet in clean_queryset]
```

Comment:
NIT: same as `snippets = list(clean_queryset)` no? Both are fine I think, just a bit verbose this way in my opinion.

Reply:
Sure it makes more sense